src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -721,11 +721,74 @@ private void modifyGenerator(Generator generator, GeneratorModificationInfos mod
-        applyElementaryModifications(generator::setTargetQ, generator::getTargetQ, modificationInfos.getReactivePowerSetpoint(), subReporter, "Reactive power set point");
         applyElementaryModifications(generator::setTargetV, generator::getTargetV, modificationInfos.getVoltageSetpoint(), subReporter, "Voltage set point");
-        applyElementaryModifications(generator::setVoltageRegulatorOn, generator::isVoltageRegulatorOn, modificationInfos.getVoltageRegulationOn(), subReporter, "Voltage regulation on");
 
-        // TODO connectivity modification
+        boolean isVoltageRegulationOn = modificationInfos.getVoltageRegulationOn() != null
+                ? modificationInfos.getVoltageRegulationOn().getValue()
+                : generator.isVoltageRegulatorOn();
+        generator.setVoltageRegulatorOn(isVoltageRegulationOn);
+
+        if (isVoltageRegulationOn) {
+            Terminal terminal = null;
+            if (modificationInfos.getRegulatingTerminalId() != null
+                    && modificationInfos.getRegulatingTerminalType() != null
+                    && modificationInfos.getRegulatingTerminalVlId() != null) {
+                terminal = getTerminalFromIdentifiable(generator.getNetwork(),
+                        modificationInfos.getRegulatingTerminalId().getValue(),
+                        modificationInfos.getRegulatingTerminalType().getValue(),
+                        modificationInfos.getRegulatingTerminalVlId().getValue());
+            }
+
+            generator.setRegulatingTerminal(terminal);
+
+            if (modificationInfos.getQPercent() != null) {
+                generator.newExtension(CoordinatedReactiveControlAdderImpl.class).withQPercent(modificationInfos.getQPercent().getValue())
+                        .add();
+            }
+        } else {
+            if (modificationInfos.getReactivePowerSetpoint() != null) {
+                applyElementaryModifications(generator::setTargetQ, generator::getTargetQ, modificationInfos.getReactivePowerSetpoint(), subReporter, "Reactive power set point");
+            }
+        }
+
+        if (modificationInfos.getTransientReactance() != null
+                && modificationInfos.getStepUpTransformerReactance() != null) {
+            generator.newExtension(GeneratorShortCircuitAdder.class)
+                    .withDirectTransX(modificationInfos.getTransientReactance().getValue())
+                    .withStepUpTransformerX(modificationInfos.getStepUpTransformerReactance().getValue())
+                    .add();
+        }
+
+        if (modificationInfos.getMarginalCost() != null) {
+            generator.newExtension(GeneratorStartupAdder.class)
+                    .withMarginalCost(modificationInfos.getMarginalCost().getValue()).add();
+        }
+        boolean participate = modificationInfos.getParticipate() != null ? modificationInfos.getParticipate().getValue()
+                : modificationInfos.getDroop() != null ? true : false;
```

Comment:
I'm not a big fan of chaining ternary operators, and in this case, the last one is superfluous.
Maybe at least add comments to explain the business logic.